Fix gcc 14 build error:

../../source3/libsmb/pylibsmb.c: In function 'py_cli_mknod':
../../source3/libsmb/pylibsmb.c:2604:15: error: implicit declaration of
 function 'makedev' [-Wimplicit-function-declaration]
 2604 |         dev = makedev(major, minor);
      |               ^~~~~~~


--- a/source3/libsmb/pylibsmb.c
+++ b/source3/libsmb/pylibsmb.c
@@ -62,6 +62,10 @@ #include "libsmb/clirap.h"
 #include "librpc/rpc/pyrpc_util.h"
 #include "librpc/gen_ndr/ndr_security.h"
 
+#if defined(HAVE_MAKEDEV) && defined(SUNOS5)
+#include <sys/mkdev.h>
+#endif
+
 #define LIST_ATTRIBUTE_MASK \
 	(FILE_ATTRIBUTE_DIRECTORY|FILE_ATTRIBUTE_SYSTEM|FILE_ATTRIBUTE_HIDDEN)
 
